test/org/jetbrains/plugins/ideavim/extension/highlightedyank/VimHighlightedYankTest.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * IdeaVim - Vim emulator for IDEs based on the IntelliJ platform
+ * Copyright (C) 2003-2020 The IdeaVim authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.jetbrains.plugins.ideavim.extension.highlightedyank
+
+import com.intellij.openapi.editor.markup.RangeHighlighter
+import com.maddyhome.idea.vim.VimPlugin
+import com.maddyhome.idea.vim.command.CommandState
+import com.maddyhome.idea.vim.helper.StringHelper
+import junit.framework.Assert
+import org.jetbrains.plugins.ideavim.VimTestCase
+
+
+class VimHighlightedYankTest : VimTestCase() {
+  override fun setUp() {
+    super.setUp()
+    enableExtensions("highlightedyank")
+  }
+
+  fun `test highlighting whole line when whole line is yanked`() {
+    doTest("yy", code, code, CommandState.Mode.COMMAND, CommandState.SubMode.NONE)
+
+    assertAllHighlightersCount(1)
+    assertHighlighterRange(1, 40, getFirstHighlighter())
+  }
+
+  fun `test highlighting single word when single word is yanked`() {
+    doTest("yiw", code, code, CommandState.Mode.COMMAND, CommandState.SubMode.NONE)
+
+    assertAllHighlightersCount(1)
+    assertHighlighterRange(5, 8, getFirstHighlighter())
+  }
+
+  fun `test removing previous highlight when new range is yanked`() {
+    configureByJavaText(code)
+    typeText(StringHelper.parseKeys("yyjyy"))
+
+    assertAllHighlightersCount(1)
+    assertHighlighterRange(40, 59, getFirstHighlighter())
+  }
+
+  fun `test removing previous highlight when entering insert mode`() {
+    doTest("yyi", code, code, CommandState.Mode.INSERT, CommandState.SubMode.NONE)
```

Comment:
May I ask you to add `@TestWithoutNeovim(reason = SkipNeovimReason.PLUGIN)` annotation to all tests that use `doTest` function? We've recently started experimenting with neovim in testing, but it doesn't support plugins at the moment.

Reply:
done